labs/b10.md

```diff
@@ -0,0 +1,152 @@
+```
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDY1qy6OE8M1gZWhyBsjE66ip+3mQpDWhP7LvT2uBloHqY6s/AkuB9Ao3hbLkJuKPghDII4UOH4DGzvcBypTjh5ClxHRMY3mInTJ6OSOuyF1lpw3SYkbIWlVWkyUABqUkt4pgWuiK6gz8egNtOo4XWf2254D+HGuYxHmLq+CInQcdqU9Wy8xmxMaq4RVqDnlmItHW6EZb4b09I87Z5T4mnc148d8p/cZ2oKXWC20EOyg55UVeTVTZw4ETzDXFFL1rOQPiMMjVfnXqoubsUCBjrctryQcxCLL3m3bOEHexAiRYmD5ItR4EiEBxiWaglBMu6OhcVYDHOQ2MEnGFbuhqXH test@test
+```
+
+After the key has been added, you'll need to add, commit, and push the file to the Gitolite server through Git.
+Assuming you are in `keydir/`, run `git add decal-checkoff.pub` to track the file under Git, `git commit` with a message describing the changes made in the commit,
+and finally `git push origin` to push the changes up to the Gitolite server.
+
+#### Adding a new Git repository
+
+Now that you've added a new Gitolite user, you'll need to create a new Git repository that they can access.
+In Gitolite, this is done by modifying the `conf/gitolite.conf` file within the `gitolite-admin` repository.
+
+Edit `gitolite.conf` to create a new repository with name `decal-checkoff` with `RW+` access given to both `admin` (you) and `decal-checkoff`.
+Hopefully, you can figure out what format Gitolite expects by looking at the entires for the `admin` and `testing` repositories.
+The only gotcha is that multiple users have to separated by a space and nothing else.
+
+Once you have done that, follow the same Git workflow as with `decal-checkoff.pub` to add, commit, and push the changes to `gitolite.conf`
+to the Gitolite server.
+
+If you've done everything correctly, running `ssh gitolite3@[your VM's address] info` should display something like:
+```
+hello admin, this is gitolite3@test running gitolite3 3.6.6-1 (Debian) on git 2.11.0
+
+ R W	decal-checkoff
+ R W	gitolite-admin
+ R W	testing
+```
+
+However, please make sure that you have added the `decal-checkoff` user and given it permissions for the `decal-checkoff` repository,
+as we will be using the `ssh info` command to check you off for the lab, and the command only works if the `decal-checkoff` user has been
+added properly.
+
+## Backups
+
+We'll be briefly practicing using `rsync`, a powerful file transfer tool often used in backup scenarios. It is difficult to 
+overstate the degree to which `rsync` and the rsync delta-transfer algorithm are used in the real world.
+
+On a superficial level, `rsync` works similarly to `scp` - it copies files from `src` to `dst`, optionally over the network,
+through an SSH tunnel. What makes `rsync` in particular so powerful is its ability to keep two directories synchronized
+while transferring little beyond the absolute difference in files between the two directories. It does this by calculating 
+the differences between blocks in a file, and only transferring the difference (the 'delta').
```

Comment:
Nit: rsync usually takes shortcuts when determining differences between blocks. For example, by default, rsync will look at the last modified time of files, and if a file isn't newer, it'll assume it wasn't changed.